src/hooks/useClientScriptLoad.ts

```diff
@@ -0,0 +1,82 @@
+import { useState, useMemo, useEffect } from "react"
+import { isNostoLoaded } from "../components/helpers"
+import type { NostoClient } from "../types"
+import type { NostoProviderProps } from "../components/NostoProvider"
+
+type NostoScriptProps = Pick<NostoProviderProps, "account" | "host" | "shopifyMarkets">
+
+export function useClientScriptLoad(props: NostoScriptProps) {
+  const { host, account, shopifyMarkets } = props
+  const [clientScriptLoadedState, setClientScriptLoadedState] = useState(false)
+  const clientScriptLoaded = useMemo(() => clientScriptLoadedState, [clientScriptLoadedState])
+
+  useEffect(() => {
+    const scriptOnload = () => {
+      if ("nostoReactTest" in window) {
+        window.nosto?.reload({
+          site: "localhost"
+        })
+      }
+      setClientScriptLoadedState(true)
+    }
+
+    // Create script element
+    function createScriptElement(urlPartial: string) {
+      const scriptEl = document.createElement("script")
+      scriptEl.type = "text/javascript"
+      scriptEl.src = `//${(host || "connect.nosto.com")}${urlPartial}`
+      scriptEl.async = true
+      scriptEl.setAttribute("nosto-client-script", "")
+      return scriptEl
+    }
+
+    // Load Nosto API stub
+    if (!window.nostojs) {
+      window.nostojs = (cb: (api: NostoClient) => void) => {
+        (window.nostojs.q = window.nostojs.q || []).push(cb)
+      }
+      window.nostojs(api => api.setAutoLoad(false))
+    }
+
+    // Load Nosto client script if not already loaded externally
+    if (!isNostoLoaded() && !shopifyMarkets) {
+      const urlPartial = `/include/${account}`
+      const script = createScriptElement(urlPartial)
```

Comment:
Here, can we have a check if the client script is already loaded on page and skip this if possible? Or else we could introduce a config flag through which we could skip adding duplicate script tags

Reply:
In a separate PR also fine

Reply:
Sure, we have loadScript flag in the [WIP PR](https://github.com/Nosto/nosto-react/pull/95/files#diff-1e68849bba5d7e4b3a74f844d6bc8d96604019ab29ae0267ef2b93db101ac5cdR44) waiting to be transferred in the next PR.

Reply:
there is already a loaded check (isNostoLoaded) and the flag will be introduced in a later PR